packages/web/src/store/tipping/utils.ts

```diff
@@ -0,0 +1,187 @@
+import { ID } from 'common/models/Identifiers'
+import { UserTip } from 'common/models/Tipping'
+import { Nullable } from 'common/utils/typeUtils'
+import { FEED_TIP_DISMISSAL_TIME_LIMIT } from 'utils/constants'
+
+const RECENT_TIPS_KEY = 'recent-tips'
+
+type RecentTipsStorage = {
+  dismissed: boolean
+  minSlot: number
+  lastDismissalTimestamp: Nullable<number>
+}
+
+const getRecentTipsStorage = () => {
+  if (!window.localStorage) {
+    return null
+  }
+
+  const value = window.localStorage.getItem(RECENT_TIPS_KEY)
+  if (!value) {
+    return null
+  }
+
+  return JSON.parse(value) as RecentTipsStorage
+}
+
+export const getMinSlotForRecentTips = () => {
+  const storage = getRecentTipsStorage()
+  return storage ? storage.minSlot : null
+}
+
+export const getDismissedSlotForRecentTips = () => {
+  const storage = getRecentTipsStorage()
+  return storage ? storage.dismissed : null
+}
+
+export const checkRecentTip = ({
```

Comment:
want to refactor this, feels like there is room to simplify / reduce lines

Reply:
If we wanted to refactor, LMK if I'm missing something (very possible), but I think we could do something like:
- In all cases, do your two checks for own tip > minSlot, then fall back to any tip > minSlot. 
- If the above is null, then logic branch on whether we're in dismissed mode or not

Reply:
there is an ever-so-slight diff with that algo, but i dont think it matters significantly. will go with it